.azure-pipelines/ultimate-pipeline.yml

```
@@ -1105,6 +1108,205 @@ stages:
       condition: succeededOrFailed()
       continueOnError: true
 
+- stage: asan_tests_windows_profiler
+  condition: > 
+    and(
+      succeeded(), 
+      eq(variables['isBenchmarksOnlyBuild'], 'False'),
+      eq(dependencies.generate_variables.outputs['generate_variables_job.generate_variables_step.IsProfilerChanged'], 'True')
+    )
+  dependsOn: [master_commit_id, generate_variables, build_windows_tracer, build_windows_profiler]
+  variables:
+    masterCommitId: $[ stageDependencies.master_commit_id.fetch.outputs['set_sha.master']]
+  jobs:
+  - template: steps/update-github-status-jobs.yml
+    parameters:
+      jobs: [Win]
+
+  - job: Win
+    pool:
+      vmImage: windows-2019
+    timeoutInMinutes: 60 #default value
+    strategy:
+      matrix:
+        x86:
+          platform: x86
+        x64:
+          platform: x64
+    variables:
+      USE_NATIVE_LOADER: true
+    
+    # Enable the Datadog Agent service for this job
+    services:
+      dd_agent: dd_agent
+
+    steps:
+    - template: steps/install-dotnet-sdks.yml
+
+    - task: DownloadPipelineArtifact@2
+      displayName: Download windows profiler home
+      inputs:
+        artifact: windows-profiler-binaries
+        path: $(System.DefaultWorkingDirectory)/profiler/_build
+
+    - template: steps/restore-working-directory.yml
+      parameters:
+        artifact: build-windows-working-directory
```

I think you should do these the other way around, otherwise you may overwrite the binaries you download.

```suggestion
    - template: steps/restore-working-directory.yml
      parameters:
        artifact: build-windows-working-directory

    - task: DownloadPipelineArtifact@2
      displayName: Download windows profiler home
      inputs:
        artifact: windows-profiler-binaries
        path: $(System.DefaultWorkingDirectory)/profiler/_build
```